langchain/src/retrievers/tests/vectara.int.test.ts

```diff
@@ -0,0 +1,148 @@
+    They are primarily found in the Southern Hemisphere, particularly in Antarctica, but species can also be found in 
+    South Africa, South America, New Zealand, and the Galapagos Islands. Penguins are expert swimmers, using their 
+    flippers for propulsion and their feet as rudders, and they can dive deep into the ocean to catch their diet of fish, 
+    squid, and krill.`;
+
+  const turtle = `Turtles are cold-blooded reptiles identifiable by their bony or cartilaginous shell developed from their 
+    ribs. They are one of the oldest and most primitive groups of reptiles and have been on Earth for more than 200 million 
+    years. Found in various habitats from deserts to marine environments around the globe, turtles are known for their 
+    slow movement on land, longevity, and often complex mating and nesting behaviors.`;
+
+  const documents = [
+    new Document({
+      pageContent: lion,
+      metadata: {
+        document_id: "1",
+        title: "Lion",
+        category: "Mammal",
+        can_fly: false,
+      },
+    }),
+    new Document({
+      pageContent: flyingSquirrel,
+      metadata: {
+        document_id: "2",
+        title: "Flying Squirrel",
+        category: "Mammal",
+        can_fly: true,
+      },
+    }),
+    new Document({
+      pageContent: penguin,
+      metadata: {
+        document_id: "3",
+        title: "Penguin",
+        category: "Bird",
+        can_fly: false,
+      },
+    }),
+    new Document({
+      pageContent: turtle,
+      metadata: {
+        document_id: "4",
+        title: "Turtle",
+        category: "Reptile",
+        can_fly: false,
+      },
+    }),
+  ];
+  return documents;
+};
+
+beforeAll(async () => {
+  // Upload the documents to Vectara
+  const store = new VectaraStore({
+    customerId: VECTARA_CUSTOMER_ID,
+    corpusId: VECTARA_CORPUS_ID,
+    apiKey: VECTARA_API_KEY,
+  });
+
+  await store.addDocuments(getdoc());
+});
+
+test("Vectara Retriever (with filters)", async () => {
+  const store = new VectaraStore({
+    customerId: VECTARA_CUSTOMER_ID,
+    corpusId: VECTARA_CORPUS_ID,
+    apiKey: VECTARA_API_KEY,
+  });
+
+  const retriever = new VectaraRetriever(store);
+  const mammals = await retriever.getRelevantDocuments("animal", 5, {
+    filter: "doc.category = 'Mammal'",
+  });
+  // expect all mammals results to contain either Lion or Flying Squirrel
+  expect(
+    mammals[0].pageContent.includes("Lion") ||
+      mammals[0].pageContent.includes("FLying Squirrel")
+  ).toBeTruthy();
+
+  const flyingAnimals = await retriever.getRelevantDocuments("animal", 5, {
+    filter: "doc.can_fly = true",
+  });
+
+  const flyingMammals = await retriever.getRelevantDocuments("animal", 5, {
+    filter: "doc.category = 'Mammal' AND doc.can_fly = true",
+  });
+  // expect all flyingMammals results to be same as flyingAnimals results
+  expect(flyingMammals).toEqual(flyingAnimals);
+
+  const reptileLion = await retriever.getRelevantDocuments("lion", 1, {
```

Comment:
I don't see what the difference is b/w reptileLion and reptile - am I missing something here?

Reply:
Yes, this test is a bit confusing and might not be necessary. This test just makes sure that the filter is working. A lion is not a reptile, so if we search for a lion it shouldn't give us any lions and should only give us reptiles.

Should I keep this test?

Reply:
Either keep it but explain why it's there to avoid others getting confused. or remove / replace with some other test.

Reply:
You are right! The reptile variable should be searching for "animal" and not "lion". Apologies for the error and confusion.

Reply:
You are right! The reptile variable should be searching for "animal" and not "lion". Apologies for the error and confusion.